content/ko/docs/howto/charts_tips_and_tricks.md

```diff
@@ -0,0 +1,307 @@
+than leaving them as bare words:
+
+```yaml
+name: {{ .Values.MyName | quote }}
+```
+
+But when working with integers _do not quote the values._ That can, in many
+cases, cause parsing errors inside of Kubernetes.
+
+```yaml
+port: {{ .Values.Port }}
+```
+
+This remark does not apply to env variables values which are expected to be
+string, even if they represent integers:
+
+```yaml
+env:
+  - name: HOST
+    value: "http://host"
+  - name: PORT
+    value: "1234"
+```
+
+##'include' 함수 사용하기
+
+Go provides a way of including one template in another using a built-in
+`template` directive. However, the built-in function cannot be used in Go
+template pipelines.
+
+To make it possible to include a template, and then perform an operation on that
+template's output, Helm has a special `include` function:
+
+```
+{{ include "toYaml" $value | indent 2 }}
+```
+
+The above includes a template called `toYaml`, passes it `$value`, and then
+passes the output of that template to the `indent` function.
+
+Because YAML ascribes significance to indentation levels and whitespace, this is
+one great way to include snippets of code, but handle indentation in a relevant
+context.
+
+## 'required' 함수 사용하기
+
+Go provides a way for setting template options to control behavior when a map is
+indexed with a key that's not present in the map. This is typically set with
+`template.Options("missingkey=option")`, where `option` can be `default`,
+`zero`, or `error`. While setting this option to error will stop execution with
+an error, this would apply to every missing key in the map. There may be
+situations where a chart developer wants to enforce this behavior for select
+values in the `values.yaml` file.
+
+The `required` function gives developers the ability to declare a value entry as
+required for template rendering. If the entry is empty in `values.yaml`, the
+template will not render and will return an error message supplied by the
+developer.
+
+For example:
+
+```
+{{ required "A valid foo is required!" .Values.foo }}
+```
+
+The above will render the template when `.Values.foo` is defined, but will fail
+to render and exit when `.Values.foo` is undefined.
+
+## 'tpl' 함수 사용하기
+
+The `tpl` function allows developers to evaluate strings as templates inside a
+template. This is useful to pass a template string as a value to a chart or
+render external configuration files. Syntax: `{{ tpl TEMPLATE_STRING VALUES }}`
+
+Examples:
+
+```yaml
+# values
+template: "{{ .Values.name }}"
+name: "Tom"
+
+# template
+{{ tpl .Values.template . }}
+
+# output
+Tom
+```
+
+Rendering an external configuration file:
+
+```yaml
+# external configuration file conf/app.conf
+firstName={{ .Values.firstName }}
+lastName={{ .Values.lastName }}
+
+# values
+firstName: Peter
+lastName: Parker
+
+# template
+{{ tpl (.Files.Get "conf/app.conf") . }}
+
+# output
+firstName=Peter
+lastName=Parker
+```
+
+## 이미지 풀 시크릿 생성하기
```

Comment:
Can you confirm this refers to an "image pull secret"? Auto-translate says "Creating an image full secret", and "full" is not the same as "pull".

Reply:
@bridgetkromhout 

"imagePullSecret" is a kind of secret in kubernetes concept.
This "Pull" means pulling the image from registry.
Maybe automatic translation is not correctly because korean pronunciation for between "pull" and "full" is same.

In this regard, please refer to the link in the k8s document.
- EN docs : https://kubernetes.io/docs/concepts/containers/images/
- KO docs : https://kubernetes.io/ko/docs/concepts/containers/images/#%ED%8C%8C%EB%93%9C%EC%97%90-imagepullsecrets-%EB%AA%85%EC%8B%9C